src/main/java/com/conveyal/gtfs/validator/SpeedTripValidator.java

```diff
@@ -37,18 +38,26 @@
     private static final Logger LOG = LoggerFactory.getLogger(SpeedTripValidator.class);
     public static final double MIN_SPEED_KPH = 0.5;
     private boolean allTravelTimesAreRounded = true;
-    private Set<NewGTFSError> travelTimeZeroErrors = new HashSet<>();
+    private final Set<NewGTFSError> travelTimeZeroErrors = new HashSet<>();
 
     public SpeedTripValidator(Feed feed, SQLErrorStorage errorStorage) {
         super(feed, errorStorage);
     }
 
     @Override
-    public void validateTrip(Trip trip, Route route, List<StopTime> stopTimes, List<Stop> stops, List<Location> locations) {
-        if (containsFlexLocations(stopTimes, locations)) {
-            // FLEX TODO: The validator needs to be refactored to work with flex locations. A stop time -> stop_id can
-            // either be a stop_id or a location id.
-            LOG.warn("Trip speed not validated because it contains flex locations!");
+    public void validateTrip(
+        Trip trip,
+        Route route,
+        List<StopTime> stopTimes,
+        List<Stop> stops,
+        List<Location> locations,
+        List<LocationGroup> locationGroups
+    ) {
+        if (hasLocationOrLocationGroupReferences(locations, locationGroups)) {
+            registerError(NewGTFSError.forFeed(
```

Comment:
As written, it should be `NewGTFSError.forEntity`. If it needs to be for the feed, this check needs to be at feed level (e.g. in `FlexValidator.validate()`).

Reply:
Fixed in https://github.com/conveyal/gtfs-lib/pull/347/commits/da910db5c20ff9e8f6ef9f25611876e7abaf5501